src/main/java/DAO/UserDAO.java

```diff
@@ -1,206 +1,78 @@
-            }
-        }
+    public List<User> getAll() {
+        return jdbcTemplate.query(FETCH_ALL_USERS_SQL, new UserMapper());
     }
 
     @Override
     public void save(User user) {
-        try {
-            connection = dataSource.getConnection();
-            preparedStatement = connection.prepareStatement(SAVE_USER_SQL);
-            preparedStatement.setString(1, user.getID().toString());
-            preparedStatement.setString(2, user.getName());
-            preparedStatement.setTimestamp(3, user.getCreationDate());
-            preparedStatement.setString(4, user.getRole().name());
-            preparedStatement.executeUpdate();
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
-        finally {
-            try {
-                preparedStatement.close();
-                connection.close();
-            } catch (SQLException e) {
-                throw new RuntimeException(e);
-            }
-        }
+        jdbcTemplate.update(SAVE_USER_SQL, user.getId(), user.getName(), user.getCreationDate(), user.getRole().name());
+        System.out.println(user.getRole().name());
     }
 
-    @Override
-    public Optional<User> fetchByID(UUID id)  {
-        try {
-            connection = dataSource.getConnection();
-            preparedStatement = connection.prepareStatement(FETCH_USER_BY_ID_SQL);
-            preparedStatement.setString(1, id.toString());
-            resultSet = preparedStatement.executeQuery();
-
-            while (resultSet.next()) {
-                UUID columnID = UUID.fromString(resultSet.getString("id"));
-                String columnName = resultSet.getString("name");
-                Timestamp columnCreationDate = resultSet.getTimestamp("creation_date");
-
-                switch (resultSet.getString("user_role")) {
-                    case "CLIENT" :
-                        return Optional.of(new Client(columnID, columnName, columnCreationDate));
-                    case "ADMIN" :
-                        return Optional.of(new Admin(columnID, columnName, columnCreationDate));
-                }
-
-            }
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
-        finally {
-            try {
-                resultSet.close();
-                preparedStatement.close();
-                connection.close();
-            } catch (SQLException e) {
-                throw new RuntimeException(e);
-            }
-        }
-        return Optional.empty();
+    public void deleteAll() {
+        jdbcTemplate.execute(DELETE_ALL_USERS_SQL);
+    }
+
+    public void updateID(UUID userID, UUID newID) {
+        jdbcTemplate.update(UPDATE_USER_ID, newID.toString(), userID.toString());
+
+        jdbcTemplate.update(CREATE_NEW_TICKET, UUID.randomUUID().toString(), newID.toString(), TicketType.DAY.name(), Timestamp.valueOf(LocalDateTime.now())); //Is it OKAY to work with 'Tickets' DBs through UserDAO?
```

Comment:
no, it is not good approach to work with 'Tickets' DBs through UserDAO. Instead of that you can make 2 calls to different dao classes from 1 method on service layer. This method has to be marked as `@Transactional`

Reply:
Fixed. I merged two services 'UserService' and 'TicketService' into one 'ApplicationService' - and create the necessary method on this level

Reply:
Generally, it is not good approach to unite services. Usually, it depends on business logic, and service is split in groups like `UserController - UserService - UserRepository` and `TicketController - TicketService - TicketRepository` or `UserController - OrderProcessorService - UserRepository/TicketRepository` . 
This task was just to find a way how to perform 2 queries inside 1 transaction. In some cases you can make a call to `TicketRepository` from `UserService` or `OrderProcessorService`, but not from `TicketRepository` to `user` table in db.

Currently, you can keep it as it is, but in the next HW (Spring Boot, Spring Data, MVC) you need try to follow commonly used architecture with separate classes.

Reply:
Got it